chasm/lib/scheduler/generator_tasks.go

```diff
@@ -102,7 +103,9 @@ func (g *GeneratorTaskExecutor) Execute(
 		// Once the idle timer expires, we close the component.
 		ctx.AddTask(scheduler, chasm.TaskAttributes{
 			ScheduledTime: idleExpiration,
-		}, &schedulerpb.SchedulerIdleTask{})
+		}, &schedulerpb.SchedulerIdleTask{
```

Comment:
Not to say that we have a problem today, but I would love to confirm this.
Can you please confirm what happens a scheduler is created with a schedule that doesn't have _any_ future runs? Will it just hang there and never close? Is the schedule rejected? What if that happens on update?

I typically like to make idle hooks like this more "declarative" because it's too easy to forget to trigger them.
For example this would be a great candidate for a hook that is triggered before transaction close.